dts/xtensa/intel/intel_apl_adsp.dtsi

```diff
@@ -92,6 +92,14 @@
 			reg = <0x1200 0x80>;
 			interrupts = <8 0 0>;
 			interrupt-parent = <&cavs0>;
+		     };
+
+		mailbox: mailbox@1180 {
+			compatible = "intel,intel-adsp-mailbox";
+			reg = <0x1180 0x20>;
+			interrupts = <0x7 0 3>;
+			interrupt-parent = <&cavs0>;
+			label = "IPM_0";
```

Comment:
This is a circumstance where IMHO devicetree is inappropriate.  The mailbox is a 100% pure software construction.  I think this wants to go in kconfig.

Reply:
@andyross, I think I am following like it is done for idc (although it was added after my PR)
https://github.com/zephyrproject-rtos/zephyr/blob/52992b065829d8280119fca96ab27d49e9d4b5fc/dts/xtensa/intel/intel_apl_adsp.dtsi#L89-L94
I have to say that this simplifies describing IRQ number.

Reply:
OK, maybe I'm just confused then.  I thought this was implemented on top of the IDC mechanism.  Is it a separate hardware device?  If so, we should probably have a driver for that, which can go into DT, right?  This seems all tied up with the IPM driver, which is the software protocol.

Reply:
IPM driver subsystem is for mailbox-type CPU features. The first use-case was the mailboxes that the x86 and ARC CPUs used to talk to each other on Arduino 101. I wouldn't go as far to say that it's purely a software construction.

It got named IPM for "inter processor mailbox" as the generic term "mailbox" is already used by a generic kernel object type.

Reply:
I might be missing something though, are we instantiating IPM on this HW without an actual HW mechanism for communicating to other CPUs?

Reply:
There's definitely a mechanism.  There is an "IDC" set of registers for delivering interrupts to other cores (both the host CPU and the DSP cores), and there's a shared mapping into the DSP SRAM that exists on both sides of the link.  This part works.  My confusion is that I thought the IPM was on top of this, but the DTS block above implies that it's different hardware.  I **think** that's just an alias of the existing IDC registers/IRQ, which would be wrong.  But the DSP is a big place and I'm still learning the map. :)

Reply:
I believe this is using the IPC mechanism which is used for host-to-DSP communication. IDC is for DSP-to-DSP.

Reply:
@andyross, @dcpleung explained right the difference between those IPC. In a way they should be described in the similar way in DTS.